extensions/redis-client/deployment/src/main/java/io/quarkus/redis/client/deployment/RedisClientProcessor.java

```diff
@@ -86,10 +87,12 @@ public void produceRedisClient(RedisClientRecorder recorder, ApplicationArchives
         Set<String> clientNames = new HashSet<>();
         clientNames.add(RedisClientUtil.DEFAULT_CLIENT);
 
-        IndexView indexView = applicationArchives.getRootArchive().getIndex();
-        Collection<AnnotationInstance> clientAnnotations = indexView.getAnnotations(REDIS_CLIENT_ANNOTATION);
-        for (AnnotationInstance annotation : clientAnnotations) {
-            clientNames.add(annotation.value().asString());
+        for (ApplicationArchive applicationArchive: applicationArchives.getAllApplicationArchives()) {
```

Comment:
You can also leverage the `io.quarkus.arc.deployment.BeanArchiveIndexBuildItem`.

Reply:
The same will have to be done for the mongo-extension https://github.com/quarkusio/quarkus/blob/33364c8700988293515dfb34d53f238f196ead15/extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/MongoClientProcessor.java#L119?

Reply:
@mkouba Yes, I will use the `BeanArchiveIndexBuildItem` this will remove a for-loop